packages/protocol/test/tokenomics/Tokenomics.test.ts

```diff
@@ -0,0 +1,302 @@
+import { expect } from "chai";
+import { BigNumber, ethers } from "ethers";
+import { ethers as hardhatEthers } from "hardhat";
+import { TaikoL1, TaikoL2, TkoToken } from "../../typechain";
+import { TestTkoToken } from "../../typechain/TestTkoToken";
+import { BlockMetadata } from "../utils/block_metadata";
+import Proposer from "../utils/proposer";
+import { proveBlock } from "../utils/prove";
+import createAndSeedWallets from "../utils/seed";
+import sleep from "../utils/sleep";
+import { defaultFeeBase, deployTaikoL1 } from "../utils/taikoL1";
+import { deployTaikoL2 } from "../utils/taikoL2";
+import deployTkoToken from "../utils/tkoToken";
+import sendTransaction from "../utils/transaction";
+
+describe("tokenomics", function () {
+    let taikoL1: TaikoL1;
+    let taikoL2: TaikoL2;
+    let l1Provider: ethers.providers.JsonRpcProvider;
+    let l2Provider: ethers.providers.JsonRpcProvider;
+    let l1Signer: any;
+    let l2Signer: any;
+    let proposerSigner: any;
+    let proverSigner: any;
+    let genesisHeight: number;
+    let genesisHash: string;
+    let tkoTokenL1: TestTkoToken;
+
+    beforeEach(async () => {
+        l1Provider = new ethers.providers.JsonRpcProvider(
+            "http://localhost:18545"
+        );
+
+        l1Provider.pollingInterval = 100;
+
+        const signers = await hardhatEthers.getSigners();
+        l1Signer = signers[0];
+
+        l2Provider = new ethers.providers.JsonRpcProvider(
+            "http://localhost:28545"
+        );
+
+        l2Signer = await l2Provider.getSigner(
+            (
+                await l2Provider.listAccounts()
+            )[0]
+        );
+
+        taikoL2 = await deployTaikoL2(l2Signer);
+
+        genesisHash = taikoL2.deployTransaction.blockHash as string;
+        genesisHeight = taikoL2.deployTransaction.blockNumber as number;
+
+        const { taikoL1: tL1, addressManager } = await deployTaikoL1(
+            genesisHash,
+            true,
+            defaultFeeBase
+        );
+
+        taikoL1 = tL1;
+
+        const { chainId } = await l1Provider.getNetwork();
+
+        [proposerSigner, proverSigner] = await createAndSeedWallets(
+            2,
+            l1Signer
+        );
+
+        tkoTokenL1 = await deployTkoToken(l1Signer, taikoL1.address);
+
+        await addressManager.setAddress(
+            `${chainId}.tko_token`,
+            tkoTokenL1.address
+        );
+
+        await tkoTokenL1
+            .connect(l1Signer)
+            .mintAnyone(
+                await proposerSigner.getAddress(),
+                ethers.utils.parseEther("100")
+            );
+
+        expect(
+            await tkoTokenL1.balanceOf(await proposerSigner.getAddress())
+        ).to.be.eq(ethers.utils.parseEther("100"));
+
+        // set up interval mining so we always get new blocks
+        await l2Provider.send("evm_setIntervalMining", [2000]);
```

Comment:
shall we use [`setAutoMine`](https://hardhat.org/hardhat-network/docs/reference#evm_setautomine) here? which may make the testing quicker than mining a block every 2 secs?